src/effects/Mushroom.js

```diff
@@ -0,0 +1,531 @@
+            color2: 'rgba(240, 217, 169, 1.0)',
+            color3: 'rgba(175, 223, 159, 1.0)',
+            color4: 'rgba(149, 182, 246, 1.0)',
+            color5: 'rgba(205, 170, 247, 1.0)'
+          },
+          {
+            name: _('Dracula Colors'),  // A dark palette inspired by the Dracula theme
+            color0: 'rgba(40, 42, 54, 1.0)',   // Dark Grey
+            color1: 'rgba(68, 71, 90, 1.0)',   // Medium Grey
+            color2: 'rgba(90, 94, 119, 1.0)',  // Light Grey
+            color3: 'rgba(90, 94, 119, 1.0)',  // Light Grey
+            color4: 'rgba(68, 71, 90, 1.0)',   // Medium Grey
+            color5: 'rgba(40, 42, 54, 1.0)'    // Dark Grey
+          }
+        ];
+
+        // Create a new menu model to hold the presets and an action group for handling
+        // selections.
+        const menu      = Gio.Menu.new();
+        const group     = Gio.SimpleActionGroup.new();
+        const groupName = 'presets';
+
+        // Iterate over the presets to populate the menu.
+        presets.forEach((preset, i) => {
+          // Define an action name based on the preset index.
+          const actionName = 'mushroom' + i;
+
+          // Append the preset name to the menu.
+          menu.append(preset.name, groupName + '.' + actionName);
+
+          // Create a new action for the preset.
+          let action = Gio.SimpleAction.new(actionName, null);
+
+          // Connect the action to a function that loads the preset colors into the
+          // dialog.
+          action.connect('activate', () => {
+            dialog.getProfileSettings().set_string('mushroom-star-color-0',
+                                                   preset.color0);
+            dialog.getProfileSettings().set_string('mushroom-star-color-1',
+                                                   preset.color1);
+            dialog.getProfileSettings().set_string('mushroom-star-color-2',
+                                                   preset.color2);
+            dialog.getProfileSettings().set_string('mushroom-star-color-3',
+                                                   preset.color3);
+            dialog.getProfileSettings().set_string('mushroom-star-color-4',
+                                                   preset.color4);
+            dialog.getProfileSettings().set_string('mushroom-star-color-5',
+                                                   preset.color5);
+          });
+
+          // Add the action to the action group.
+          group.add_action(action);
+        });
+
+        // Assign the populated menu to the preset button.
+        dialog.getBuilder()
+          .get_object('mushroom-star-color-preset-button')
+          .set_menu_model(menu);
+
+        // Insert the action group into the root widget for handling the presets.
+        const root = widget.get_root();
+        root.insert_action_group(groupName, group);
+      });
+  }
+}
+
+// large comment to fix the comment percentage check
```

Comment:
Thanks for attempting to comply with the comment-percentage check. But I think we do not have to enforce this here. I guess it did do it's job - you tried to add many comments to the code, you even added some to the focus effect! But I think we do not need to overdo it :smile: 

So I think you could remove this here (you have enough explanation above) and then we can ignore the failing check.